src/engine/Server.cpp

```diff
@@ -45,14 +47,21 @@ void Server::initialize(const string& ontologyBaseName, bool useText,
 
 // _____________________________________________________________________________
 void Server::run() {
+  QueryExecutionContext qec(_index, _engine);
+  std::vector<std::thread> threads;
+  for(int i = 0; i < _numThreads; ++i) {
+    threads.emplace_back(&Server::runAcceptLoop, this, &qec);
+  }
+  for(std::thread& worker: threads) {
+    worker.join();
+  }
+}
+// _____________________________________________________________________________
+void Server::runAcceptLoop(QueryExecutionContext* qec) {
   if (!_initialized) {
```

Comment:
I think the initilization could be checked in the run method, before the worker threads are spawned, to avoid posiibly printing the error message several times.